cmd/crank/beta/validate/manager_test.go

```diff
@@ -230,6 +258,96 @@ func TestConfigurationTypeSupport(t *testing.T) {
 	}
 }
 
+func TestAddDependencies(t *testing.T) {
+	cd1 := static.NewLayer(configDep1Yaml, types.OCILayer)
+	cd2 := static.NewLayer(configDep2Yaml, types.OCILayer)
+	pd1 := static.NewLayer(providerYaml, types.OCILayer)
+	fd1 := static.NewLayer(funcYaml, types.OCILayer)
+
+	fetchMockFunc := func(image string) (*conregv1.Layer, error) {
+		switch image {
+		case "config-dep-1:v1.3.0":
+			return &cd1, nil
+		case "config-dep-2:v1.3.0":
+			return &cd2, nil
+		case "provider-dep-1:v1.3.0":
+			return &pd1, nil
+		case "function-dep-1:v1.3.0":
+			return &fd1, nil
+		default:
+			return nil, fmt.Errorf("unknown image: %s", image)
+		}
+	}
+
+	type args struct {
+		extensions []*unstructured.Unstructured
+		fetchMock  func(image string) (*conregv1.Layer, error)
+	}
+	type want struct {
+		confs int
+		deps  int
+		err   error
+	}
+	cases := map[string]struct {
+		reason string
+		args   args
+		want   want
+	}{
+		"SuccessfulDependenciesAddition": {
+			// config-dep-1
+			//└─►config-dep-2
+			//config-dep-2
+			//├─►provider-dep-1
+			//└─►function-dep-1
+			reason: "All dependencies should be successfully fetched and added",
+			args: args{
+				fetchMock: fetchMockFunc,
+				extensions: []*unstructured.Unstructured{
+					{
+						Object: map[string]interface{}{
+							"apiVersion": "pkg.crossplane.io/v1alpha1",
+							"kind":       "Configuration",
+							"metadata": map[string]interface{}{
+								"name": "config-dep-1",
+							},
+							"spec": map[string]interface{}{
+								"package": "config-dep-1:v1.3.0",
+							},
+						},
+					},
+				},
+			},
+			want: want{
+				confs: 2, // 1 Base Configuration, 1 Child Configuration
+				deps:  4, // 2 Configurations, 1 provider, 1 function
```

Comment:
listing out the names of each would be a bit more helpful to understand here, e.g.:
```
// 1 Base configuration (config-dep-1), 1 child configuration (config-dep-2)
// 2 configurations (config-dep-1, config-dep-2), 1 provider (provider-dep-1), 1 function (function-dep-1)
```